### ISSUE TITLE
GetComponent call inside Bind method results in a UnityException

### ISSUE THREAD

**Issue by mauricioribeiro** [pusher/pusher-websocket-unity#27]
Hi guys,

I have the following code in my PusherManager:
```C#
channel.Bind("my-event", (dynamic data) =>
        {
               _myGameObject.GetComponent<Player>().DoSomething();
        }
});
```

Basically, I'm just trying to call a method from an attached `GameObject` in my PusherManager. But, since the code inside `Bind` method is async and it's using a different thread, the code throws this error:

```
GetComponentFastPath can only be called from the main thread.
Constructors and field initializers will be executed from the loading thread when loading a scene.
Don't use this function in the constructor or field initializers, instead move initialization code to the Awake or Start function.
```

I made some research and I think Unity blocks any `GetComponent` call inside parallel threads. Does anybody have any clue about how to solve this?

**Comment by twomedia** [comment]
We used this nice plugin on the asset store for dispatching to the MainThread: [Task Parallel](https://assetstore.unity.com/packages/tools/integration/task-parallel-82257)

Then in that Bind action you would do this:
```
UnityTask.RunOnUIThread(() => {
_myGameObject.GetComponent<Player>().DoSomething();
});
```

**Comment by mauricioribeiro** [comment]
Thank you @twomedia. After some time, I managed to solve this by setting a boolean flag variable (like `is_event_triggered`) on async `Bind` method to avoid get component on it. 

So, in the async scope, I only set this flag to true instead call component `DoSomething()` and then in my not sync`Update` scope I verify this flag. If it's true, I finally call the component`DoSomething()`.

That was a workaround, but solved my problem. Anyway, I will check this plugin :)
